internal/db/state/contract.go

```diff
@@ -0,0 +1,78 @@
+package state
+
+import (
+	"encoding/json"
+
+	"github.com/NethermindEth/juno/pkg/felt"
+	"github.com/NethermindEth/juno/pkg/state"
+	"github.com/NethermindEth/juno/pkg/types"
+	"google.golang.org/protobuf/proto"
+)
+
+func (m *Manager) GetContractState(hash *felt.Felt) (*state.ContractState, error) {
+	raw, err := m.stateDatabase.Get(hash.ByteSlice())
+	if err != nil {
+		// Database error
+		return nil, err
+	}
+	if raw == nil {
+		// Not found
+		return nil, nil
+	}
```

Comment:
As far as I am aware, the db package has an error for queries that result in not found errors, `ErrNotFound`. Shouldn't we be checking for that instead of examining whether `raw == nil`?